GAE/src/org/waterforpeople/mapping/app/web/rest/QuestionRestService.java

```diff
@@ -350,8 +349,8 @@ private static void setQuestionOptionsIdList(List<QuestionDto> questionDtoList,
             } else {
                 q = copyQuestion(questionDto);
             }
-            dto = new QuestionDto();
-            DtoMarshaller.copyToDto(q, dto);
+            QuestionDtoMapper qdm = new QuestionDtoMapper();
+            dto = qdm.transform(q);
             statusDto.setStatus("ok");
             statusDto.setMessage("");
 
```

Comment:
infact line 357 - 371 is copied and pasted in all the functions. I suggest extracting into a single function and reuse